proposals/Exceptions.md

```diff
@@ -170,21 +174,70 @@ also be sure to also pop off the caught exception values.
 
 ### Rethrows
 
-The _rethrow_ instruction has no arguments, and can only appear in catch
-blocks. It always re-throws the exception caught by the catch block. This allows
-the catch block to clean up state before the exception is passed back to the
-next enclosing try block.
+The _rethrow_ instruction can only appear in the body of a catch block.  The
+_rethrow_ instruction gets a single index argument. It is to disambiguate which
+caught exception is to be rethrown, when inside nested catch blocks (of
+corresponding nested try blocks). The blocks are numbered monotonically
```

Comment:
I found this hard to understand until I saw @rossberg-chromium's description [here](https://github.com/WebAssembly/exception-handling/issues/10#issue-228061591). Include it below?